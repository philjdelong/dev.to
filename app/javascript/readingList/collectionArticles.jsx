import { h } from 'preact';
import PropTypes from 'prop-types';
import { Article } from './components/Article';

export const CollectionArticles = collection => {
  const articles = JSON.parse(collection.articles);
  const articleList = articles.map(article => {
    return <Article {...article} />;
  });

  return <section className="collection item-wrapper">{articleList}</section>;
};

CollectionArticles.propTypes = {
  collection: PropTypes.string.isRequired,
};
