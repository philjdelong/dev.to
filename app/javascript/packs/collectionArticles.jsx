import { h, render } from 'preact';
import { getUserDataAndCsrfToken } from '../chat/util';
import { CollectionArticles } from '../readingList/collectionArticles';

function loadElement() {
  getUserDataAndCsrfToken().then(() => {
    const root = document.getElementById('collection-articles');
    if (root) {
      render(
        <CollectionArticles
          articles={root.dataset.articles}
          collection={root.dataset.collection}
        />,
        root,
        root.firstElementChild,
      );
    }
  });
}

window.InstantClick.on('change', () => {
  loadElement();
});

loadElement();
