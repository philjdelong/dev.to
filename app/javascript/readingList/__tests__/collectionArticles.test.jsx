import { h } from 'preact';
import render from 'preact-render-to-json';
import { CollectionArticles } from '../collectionArticles';

describe('<CollectionForm />', () => {
  it.skip('renders properly', () => {
    const tree = render(<CollectionArticles />);
    expect(tree).toMatchSnapshot();
  });
});
