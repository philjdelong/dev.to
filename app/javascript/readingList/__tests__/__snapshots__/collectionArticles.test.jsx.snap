// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollectionArticles /> renders properly 1`] = `
<div
  class="items-container"
>
  <section
    class="collection-cont results results--loaded"
  >
    <div
      class="results-header collection-header"
    >
      Testing Collection
      <section
        class="tag-display"
      >
        <a
          href="/t/javascript"
        >
          <span
            class="tag"
          >
             #javascript
          </span>
        </a>
      </section>
    </div>
    <section
      class="collection item-wrapper"
    >
      <div
        class="coll-item"
      >
        <a
          class="coll-title"
          href="/joeloelacey/example-test-article-29c8"
        >
          Example Test Article
        </a>
        <div
          class="coll-info"
        >
          <p>
            Reading Time: 1 Min
          </p>
          <p>
            Created: March 29
          </p>
        </div>
        <div
          class="featured-tags tags"
        >
          <section
            class="tag-display"
          >
            <a
              href="/t/javascript"
            >
              <span
                class="tag"
              >
                 #javascript
              </span>
            </a>
          </section>
        </div>
      </div>
    </section>
    <section
      class="collection item-wrapper"
    >
      <div
        class="coll-item"
      >
        <a
          class="coll-title"
          href="/joeloelacey/secondary-test-article-44c0"
        >
          Secondary Test Article
        </a>
        <div
          class="coll-info"
        >
          <p>
            Reading Time: 1 Min
          </p>
          <p>
            Created: March 29
          </p>
        </div>
        <div
          class="featured-tags tags"
        >
          <section
            class="tag-display"
          >
            <a
              href="/t/javascript"
            >
              <span
                class="tag"
              >
                 #javascript
              </span>
            </a>
          </section>
        </div>
      </div>
    </section>
  </section>
</div>
`;
